### PR TITLE
add multicall in UserProxy

### PR DESCRIPTION
**Pull request by charlesjhongc**

### CODE REVIEW

File: contracts/test/UserProxy.t.sol [hunk 1]
```diff
@@ -303,4 +304,32 @@ contract UserProxyTest is Test {
...
+     *                Test: multicall                  *
+     ***************************************************/
+
+    function testMulticallAMMandRFQ() public {
+        userProxy.upgradeAMMWrapper(address(strategy), true);
+        userProxy.upgradeRFQ(address(strategy), false);
+
+        bytes[] memory data = new bytes[](2);
+        data[0] = abi.encodeWithSelector(UserProxy.toAMM.selector, MockStrategy.execute.selector);
+        data[0] = abi.encodeWithSelector(UserProxy.toRFQ.selector, MockStrategy.execute.selector);
```
Comment (NIC619) [inline contracts/test/UserProxy.t.sol:318]:
```suggestion
        data[1] = abi.encodeWithSelector(UserProxy.toRFQ.selector, MockStrategy.execute.selector);
```

File: contracts/test/UserProxy.t.sol [hunk 2]
```diff
@@ -303,4 +304,32 @@ contract UserProxyTest is Test {
...
+        userProxy.multicall(data, false);
+    }
+
+    function testMulticallRevertOnFail() public {
+        userProxy.upgradeAMMWrapper(address(strategy), true);
+        userProxy.upgradeRFQ(address(strategy), false);
+
+        bytes[] memory data = new bytes[](2);
+        data[0] = abi.encodeWithSelector(UserProxy.toAMM.selector, MockStrategy.execute.selector);
+        data[0] = abi.encodeWithSelector(UserProxy.toRFQ.selector, MockStrategy.execute.selector);
```
Comment (NIC619) [inline contracts/test/UserProxy.t.sol:330]:
```suggestion
        data[1] = abi.encodeWithSelector(UserProxy.toRFQ.selector, MockStrategy.execute.selector);
```

File: contracts/test/UserProxy.t.sol (hunk 1 above)
Comment (charlesjhongc) [reply contracts/test/UserProxy.t.sol]: Fixed in f7b4d56

File: contracts/test/UserProxy.t.sol (hunk 2 above)
Comment (charlesjhongc) [reply contracts/test/UserProxy.t.sol]: Fixed in f7b4d56

**Review by changwu-tw — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Review by NIC619 — APPROVED** [review]

**Pull request merged**